hw3/lambda_function.py

```diff
@@ -0,0 +1,53 @@
+import json
+import boto3
+
+
+def lambda_handler(event, context):
+    try:
+        dynamodb = boto3.resource('dynamodb')
+        table = dynamodb.Table('my-table')
+        print(event)
+        headers = event['headers']
+        request_id = headers.get('request-id')
+        print(request_id)
+        if not request_id:
+            return {
+                'statusCode': 400,
+                'body': 'Missing request-id header'
+            }
+
+        response = {
+            'request-id': request_id,
+            'message': 'Hello, world1!'
+        }
+        s3 = boto3.client('s3')
+        bucket_name = 'my-first-bucket-ihor2'
+        file_key = f'{request_id}.json'
+
+        try:
+            s3.put_object(
+                Body=json.dumps(response),
+                Bucket=bucket_name,
+                Key=file_key
+            )
+            table.put_item(
+                Item={
+                    'id': request_id,
+                    'message': 'Hello, world!'
+                }
+            )
```

Comment:
It looks like you misunderstood the task. Altogether you should have three functions:
1. triggered by HTTP request and stores event to event-stream
2. reads from event-stream, writes to s3
3. reads from event-stream, writes to a database

Please check the diagram in the task description: https://github.com/v1r7u/cloud_computing_course/blob/main/homeworks/04_db.md